pkg/utils/secrets/etcd_encryption_key.go

```diff
@@ -6,13 +6,18 @@ package secrets
 
 import (
 	"fmt"
+
+	"github.com/gardener/gardener/pkg/utils"
 )
 
 const (
 	// DataKeyEncryptionKeyName is the key in a secret data holding the key.
 	DataKeyEncryptionKeyName = "key"
 	// DataKeyEncryptionSecret is the key in a secret data holding the secret.
 	DataKeyEncryptionSecret = "secret"
+	// DataKeyEncryptionSecretEncoding is the key in a secret data that defines if the secret is already base64 encoded or not.
+	// Possible values are "base64" and "none".
```

Comment:
Currently the "base64" value isn't checked anywhere so this is a bit misleading. What about adding info that if the key is missing from the secret or its value is empty that `base64` is the default value.

Reply:
Yes, this is a good suggestion. Have in mind that also this is not entirely true, because a generated key with length of 31 would still be invalid `base64` encoding because of the missing padding.

Should we rather just mention that the only possible value is "none" and accept that we had a bug, which we are now addressing.

Reply:
Yeah you're right. Let's mention that the possible value currently is `none` and when no encoding is specified it was due to old/buggy behaviour that still has to be supported by directly using the secret data.